### PR TITLE
CASMPET-6079: Add Keycloak version 16.1.1

### PR DESCRIPTION
**Pull request by ndavidson-hpe**
Added Keycloak version 16.1.1 for use in the first major jump in Keycloak version

* Resolves [CASMPET-6079](https://jira-pro.its.hpecorp.net:8443/browse/CASMPET-6079)
* Future work in [CASMPET-6079](https://jira-pro.its.hpecorp.net:8443/browse/CASMPET-6079)

### CODE REVIEW

**Comment by rnoska-hpe** [comment]
I think you'll need to suppress the Snyk errors.
```
Error: Vulnerabilities are reported by snyk scan. Set 'fail_on_snyk_errors' input parameter to 'false' to suppress this error.
Error: Process completed with exit code 1.
```

File: .github/workflows/docker.io.jboss.keycloak.16.1.1.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#
+# MIT License
+#
+# (C) Copyright [2022] Hewlett Packard Enterprise Development LP
```
Comment (bo-quan) [inline .github/workflows/docker.io.jboss.keycloak.16.1.1.yaml:4]:
```suggestion
# (C) Copyright [2023] Hewlett Packard Enterprise Development LP
```

File: docker.io/jboss/keycloak/16.1.1/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#
+# MIT License
+#
+# (C) Copyright [2022] Hewlett Packard Enterprise Development LP
```
Comment (bo-quan) [inline docker.io/jboss/keycloak/16.1.1/Dockerfile:4]:
```suggestion
# (C) Copyright [2023] Hewlett Packard Enterprise Development LP
```

**Review by bo-quan — CHANGES_REQUESTED** [review]

**Review by bo-quan — DISMISSED** [review]

File: .github/workflows/docker.io.jboss.keycloak.16.1.1.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          context_path: ${{ env.CONTEXT_PATH }}
+          docker_repo: ${{ env.DOCKER_REPO }}
+          docker_tag: ${{ env.DOCKER_TAG }}
+          artifactory_algol60_token: ${{ secrets.ARTIFACTORY_ALGOL60_TOKEN }}
+          cosign_gcp_workload_identity_provider: ${{ secrets.COSIGN_GCP_WORKLOAD_IDENTITY_PROVIDER }}
+          cosign_gcp_service_account: ${{ secrets.COSIGN_GCP_SERVICE_ACCOUNT }}
+          cosign_key: ${{ secrets.COSIGN_KEY }}
+          snyk_token: ${{ secrets.SNYK_TOKEN }}
+          github_sha: $GITHUB_SHA
+          fail_on_snyk_errors: true
```
Comment (bo-quan) [inline .github/workflows/docker.io.jboss.keycloak.16.1.1.yaml:68]: Rob is correct that you want to change this to false

**Review by rnoska-hpe — APPROVED** [review]

**Pull request merged**